diff --git a/lib/get-exports.js b/lib/get-exports.js
index 9a78e8c6effb730d32ba768b4df2e80568356b37..de88a37b5e315bd591361b4e88fc7045a08bda64 100644
--- a/lib/get-exports.js
+++ b/lib/get-exports.js
@@ -43,6 +43,9 @@ async function getCjsExports (url, context, parentLoad, source) {
           full.add(each)
         }
       } else {
+        if (re === '.') {
+          re = './'
+        }
         // Resolve the re-exported module relative to the current module.
         const newUrl = pathToFileURL(require.resolve(re, { paths: [dirname(fileURLToPath(url))] })).href
         for (const each of await getExports(newUrl, context, parentLoad)) {
